flink-ai-flow/ai_flow/test/endpoint/test_java_client.py

```diff
@@ -0,0 +1,194 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+import os
+import os
+import subprocess
+import unittest
+
+from ai_flow.endpoint.server.server import AIFlowServer
+from ai_flow.store.db.base_model import base
+from ai_flow.test.store.test_sqlalchemy_store import _get_store
+from notification_service.event_storage import DbEventStorage
+
+_SQLITE_DB_FILE = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'aiflow.db')
+_SQLITE_DB_URI = '%s%s' % ('sqlite:///', _SQLITE_DB_FILE)
+_CONFIG_FILE_PATH = '../../java/client/src/test/resources/test.properties'
+
+
+class JavaAIFlowClientTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        port = cls._get_properties(_CONFIG_FILE_PATH, 'port')
+        if os.path.exists(_SQLITE_DB_FILE):
+            os.remove(_SQLITE_DB_FILE)
+        cls.server = AIFlowServer(store_uri=_SQLITE_DB_URI, port=port, start_scheduler_service=False)
+        cls.server.run()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.server.stop()
+        os.remove(_SQLITE_DB_FILE)
+
+    def setUp(self) -> None:
+        _get_store(_SQLITE_DB_URI)
+
+    def tearDown(self) -> None:
+        store = _get_store(_SQLITE_DB_URI)
+        base.metadata.drop_all(store.db_engine)
+        storage = DbEventStorage(db_conn=_SQLITE_DB_URI, create_table_if_not_exists=False)
+        storage.clean_up()
+
+    @staticmethod
+    def _get_properties(file_path: str, key: str):
+        try:
+            pro_file = open(file_path, 'r')
+            properties = {}
+            for line in pro_file:
+                if line.find('=') > 0:
+                    strs = line.replace('\n', '').replace(' ', '').split('=')
+                    properties[strs[0]] = strs[1]
+            print(properties)
+            return properties[key]
+        except Exception as e:
+            raise e
+        finally:
+            pro_file.close()
+
+    @staticmethod
+    def _run_test_with_java_client(method):
+        jar_path = 'java/client/target/aiflow-client-1.0-SNAPSHOT-jar-with-dependencies.jar'
```

Comment:
IIUTC, the `_run_test_with_java_client` method replies on the Java Maven package. But the python tests don't need to couple with Java Maven. If the version of Java Maven project updates, does this work well?

Reply:
You're right and I've change it to use wildcard to find  *jar-with-dependencies.jar